api/src/pcapi/scripts/full_index_offers.py

```diff
@@ -61,16 +60,21 @@ def full_index_offers(start, end):  # type: ignore [no-untyped-def]
     """
     if start > end:
         raise ValueError('"start" must be less than "end"')
-    backend = algolia.AlgoliaBackend()
+    backend = search.backends.algolia.AlgoliaBackend()
 
     queue = []
 
-    def enqueue_or_index(q, offer, force_index=False):  # type: ignore [no-untyped-def]
-        if offer:
-            q.append(offer)
+    def enqueue_or_index(
+        q: list,
+        offer: offers_models.Offer | None,
+        last_30_days_bookings: dict[int, int] | None,
+        force_index: bool = False,
+    ) -> None:
+        if offer and last_30_days_bookings is not None:
+            q.append((offer, last_30_days_bookings.get(offer.id) or 0))
         if force_index or len(q) > BATCH_SIZE:
             try:
-                backend.index_offers(q)
+                backend.index_offers([data[0] for data in q], {data[0].id: data[1] for data in q})
```

Comment:
(style) On peut unpack `data` et avoir un meilleur nommage : 

```suggestion
                backend.index_offers([offer for offer, _ in q], {offer.id: n_bookings for offer, n_bookings in q})
```